pkg/cmd/server/api/types.go

```diff
@@ -46,8 +46,11 @@ type NodeConfig struct {
 	// DNSIP holds the IP
 	DNSIP string
 
-	// NetworkPluginName is a string specifying the networking plugin
-	NetworkPluginName string
+	// Deprecated and maintained for backward compatibility, use NodeNetworkConfig.PluginName instead
+	DeprecatedNetworkPluginName string
```

Comment:
you don't need the deprecated field in the internal version, just in the v1 struct